lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -212,16 +217,7 @@ def store(payment_method, options = {})
           result = add_payment_method_token(params, payment_method, options)
           return result if result.is_a?(ActiveMerchant::Billing::Response)
 
-          if options[:customer]
-            customer_id = options[:customer]
-          else
-            post[:description] = options[:description] if options[:description]
-            post[:email] = options[:email] if options[:email]
-            options = format_idempotency_key(options, 'customer')
-            post[:expand] = [:sources]
-            customer = commit(:post, 'customers', post, options)
-            customer_id = customer.params['id']
-          end
+          customer_id = options[:customer] || customer(post, payment_method, options).params['id']
```

Comment:
There are some fields being mutated with `post` here when creating a customer, why delete them? Am I missing where `email` or the re-formatted `idempotency_key` would now get added to `post`?

Reply:
`email` get set in `add_address` for `billing_address` and [idempotency_key here](https://github.com/activemerchant/active_merchant/pull/4782/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbR242).  I just moved the logic to [customer method](https://github.com/activemerchant/active_merchant/pull/4782/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbR230) because that's the only place that post get's used.

Reply:
`email` can be sent in the top level `options[:email]` not just in the nested `address` hash and the current code of `add_address` is only run if an address is passed in. Would this mean that there is a scenario where someone passes in an `email` at the top level, no address, and the email does not get sent?

Reply:
This is true, I'll update the email